Ginger/GingerCoreTest/Misc/DataSourceTest.cs

```diff
@@ -0,0 +1,264 @@
+            accessDataSource.FileFullPath = TestResources.GetTestResourcesFile(@"DataSources\GingerDataSource.mdb");
+        }
+
+        [TestMethod]
+        public void GetTableList()
+        {
+            //Act 
+            sourceTables = accessDataSource.GetTablesList();
+
+            //Assert
+            Assert.AreNotEqual(sourceTables.Count, 0);
+        }
+
+        [TestMethod]
+        public void AddTableAndDeleteTable()
+        {
+            //Act : add table
+            accessDataSource.AddTable("NewTableAs", "[GINGER_ID] AUTOINCREMENT,[GINGER_USED] Text,[GINGER_LAST_UPDATED_BY] Text,[GINGER_LAST_UPDATE_DATETIME] Text");
+
+            //Assert
+            Assert.AreEqual(accessDataSource.IsTableExist("NewTableAs"), true);
+
+            //Act : delete table
+            accessDataSource.DeleteTable("NewTableAs");
+
+            //Assert
+            Assert.AreEqual(accessDataSource.IsTableExist("NewTableAs"), false);
+        }
+
+        [TestMethod]
+        public void AddAndDeleteColunmfromTable()
+        {
+            //Arrange
+            List<string> ColunmList;
+            bool IsColunmExists = false;
+
+            //Act :add colunm
+            accessDataSource.AddColumn("MyCustomizedDataTable", "New", "Text");
+            ColunmList = accessDataSource.GetColumnList("MyCustomizedDataTable");
+            IsColunmExists = ColunmList.Any(s => s.Contains("New"));
+
+            //Assert
+            Assert.AreEqual(IsColunmExists, true);
+
+            //Act : delete colunm
+            accessDataSource.RemoveColumn("MyCustomizedDataTable", "New");
+            ColunmList = accessDataSource.GetColumnList("MyCustomizedDataTable");
+            IsColunmExists = ColunmList.Any(s => s.Contains("New"));
+            //Assert
+            Assert.AreEqual(IsColunmExists, false);
+        }
+
+        [TestMethod]
+        public void RenameTable()
+        {
+            //Arrange
+            string NewName = "TestRename";
+
+            //Act
+            accessDataSource.RenameTable("MyCustomizedDataTable", NewName);
+            Assert.AreEqual(accessDataSource.IsTableExist(NewName), true);
+            accessDataSource.RenameTable(NewName, "MyCustomizedDataTable");
+
+            //Assert
+            Assert.AreEqual(accessDataSource.IsTableExist("MyCustomizedDataTable"), true);
+        }
+
+        [TestMethod]
+        public void SaveTable()
+        {
+            // Arrange
+            ObservableList<DataSourceTable> dataSourceTableList = accessDataSource.GetTablesList();
+            DataSourceTable dataSource = null;
+            List<string> mColumnNames = null;
+            foreach (DataSourceTable dataSourceTable in dataSourceTableList)
+            {
+                if (dataSourceTable.Name == "MyCustomizedDataTable")
+                {
+                    dataSource = dataSourceTable;
+                }
+            }
+            int rowsCount = accessDataSource.GetRowCount(dataSource.Name);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['dataSource' is null on at least one execution path.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4855725)